### ISSUE TITLE
Avoid using root path on admin UI

### ISSUE THREAD

**Issue by kureikain** [influxdb/influxdb#1236]
I want to discuss this change. If you guys are ok, I'm happy for a PR. 

Currently, admin UI on host:8083 use absolute asset path. On some strict environment, I cannot open port 8083, I therefore using Nginx and set it up like this: http://host/influxdb_admin -> 127.0.0.1:8083. However, the asset cannot be loaded because they will be set to host:8083/asset. Note that, port 8083 isn't open to public.

I suggest that we removed that and respect that uri and prepend them to asset path propertly to make above setting work.

**Comment by begriffs** [comment]
+1

I would like to proxy on a nested root path as well.

**Comment by Sean1e** [comment]
guys, do you have a plan to introduce this functionality within the next major release 0.9 version?

**Comment by Sean1e** [comment]
Could anyone help to solve authentication issue? It goes to  http://IP:8086/cluster_admins rather than http://IP:80/influxdb_web/cluster_admins (we defined /influxdb_web as Nginx custom location)

**Comment by mikhailov** [comment]
guys, is there any option to use Nginx as a reverse-proxy for InfluxDB web/api access and avoid using CORS?

**Comment by pauldix** [comment]
not currently. We'll see what we can do for 0.9.0. Out of curiosity, why do
you want to use Nginx as a reverse proxy?

On Tue, Jan 20, 2015 at 1:16 PM, Anatoly Mikhailov <notifications@github.com

**Comment by xrl** [comment]
Nginx is great for having uniform access controls (LDAP integration) and also encapsulating deployment complexity. I make a landing page which links to /graphite, /grafana/, /kibana, and now I'd like to have /influxdb. Nginx in turn hides how all those are services are running.

**Comment by kureikain** [comment]
@pauldix I totally agree with @xrl 
Also, we cannot open other port than port 80 bcuz a strict firewall and have to use HAProxy/Nginx then detect the hostname and route into correct backend. It's much more secure than IMHO

**Comment by mikhailov** [comment]
yes, @xrl and @kureikain are absolutely right, by some security concern (security audit requires to have 80/443 ports open only for app nodes, for instance), we can keep SSL certificates settings in one place (Nginx configuration), we can load balance between machines to build failover for influxDB web interface and the API. Many many different use cases where reverse-proxy makes sense.

**Comment by porjo** [comment]
This is a duplicate of https://github.com/influxdb/influxdb-admin/issues/34. I will close that issue in favour of this one.

**Comment by mikhailov** [comment]
@porjo, will you merge the commit https://github.com/Sean1e/influxdb/commit/fe0967c6ab5d672ec896a3120cd0ef668e4258ff ?

**Comment by porjo** [comment]
@mikhailov I'm not sure what you mean (I'm not a maintainer BTW).

**Comment by dhain** [comment]
+1

**Comment by BartVB** [comment]
Same problem here. I'm using nginx in front of InfluxDB for the same reasons as @xrl This is my current config:

https://github.com/influxdb/influxdb/issues/1205#issuecomment-76965477

This works except for the InfluxDB admin because I can't tell it to use a different root and no username/password.

Another related issue is #1621

**Comment by mikhailov** [comment]
Sean1e@fe0967c commit makes static assets URI relative which is an answer, but it seems influxdb will have admin panel as a separate tool now?

**Comment by gillesdemey** [comment]
+1, is anyone making an effort to fix this? Still having this issue in RC10.

**Comment by bitdivine** [comment]
+1

**Comment by bitdivine** [comment]
Here is a hacky way that will let you run a reverse proxy through port 80 on apache:
https://gist.github.com/bitdivine/3366f3d21e025cac3b83
This is tested with the latest deb package from the influxdb web site at the time of writing.

By the way, how do you plan to implement support for reverse proxying?  Do you know what the solution is likely to look like?  For example you could do this by setting a prefix in the config file:
[admin]
port   = 8083  <-- the actual port
public_prefix = ':8080/admin/' <-- what is shoved onto the front of links in web pages.
The proxy software is then responsible for removing the prefix.

**Comment by Gurado** [comment]
:+1:

**Comment by woodsae** [comment]
:+1:

**Comment by toddboom** [comment]
Once #2746 is merged, all assets should be using relative paths. I believe that resolves the original issue.

If the reverse-proxy support is still needed, could someone open up a new issue for that requirement?

**Comment by mikhailov** [comment]
:+1:
